live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/filter/AspectFilterApplierTest.java

```diff
@@ -243,4 +243,21 @@ void shouldFilterOutWhenEmptyAllowedAndAspectInDenied()
         // then
         assertFalse(result);
     }
+
+    @Test
+    void shouldNotFilterOutWhenAspectsPartiallyInAllowedAndAspectNotInDenied()
+    {
+        final String aspect1 = "cm:aspect1";
+        final String aspect2 = "cm:aspect2";
+        final String aspect3 = "cm:aspect3";
+        given(mockResource.getAspectNames()).willReturn(Set.of(aspect1));
+        given(mockAspect.allow()).willReturn(List.of(aspect1, aspect2));
+        given(mockAspect.deny()).willReturn(List.of(aspect3));
+
+        // when
+        boolean result = objectUnderTest.applyFilter(mockRepoEvent, mockFilter);
+
+        // then
+        assertFalse(result);
```

Comment:
This should be `assertTrue` because the node has `aspect1 OR aspect2 AND NOT aspect3`